### ISSUE TITLE
Add `shuffle` arg to initial_split

### ISSUE THREAD

**Issue by ClaytonJY** [tidymodels/rsample#53]
I'm a big fan of `initial_split()`; feels very natural to make one "big" split, then e.g. k-fold the training set of the big split.

Would it make sense to allow for ordered/non-shuffling samples, for use in timeseries problems where further sampling would be done by `rolling_origin`?

The easiest solution is probably to add an arg like `shuffle = FALSE` to `initial_split`, which would mirror how `scikit-learn` does it (http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html).

It's easy to imagine going deeper, allowing for e.g. the `skip` arg of `rolling_origin`, or an `order_by` (similar to https://github.com/tidymodels/recipes/issues/171), but starting small with a `shuffle` arg might be a good start.

**Comment by topepo** [comment]
I don't think that this is needed since we use random selection to make splits. Looking at their docs, they don't so shuffling gives them random selection.

**Comment by ClaytonJY** [comment]
@topepo but how can we _avoid_ shuffling when making a sample, which is necessary for time-dependent sampling? One option would be a new function entirely, or we could add a `shuffle` arg to `initial_split` that is `TRUE` by default (fully backwards compatible), but can be set to `FALSE` to take the _first_ x% as training instead of a random x%

**Comment by topepo** [comment]
If it is only that function, then an additional option would work (and could that an alternate code path internally to do so). Otherwise, I'd rather create new functions to do non-random sampling (like `rolling_origin`).

**Comment by ClaytonJY** [comment]
I guess it depends where you see this going; if you think we'll eventually need a more fully-featured time-dependent initial sampling function, we should probably start with a new function now, otherwise my vote is for a `shuffle` arg.

The more I think about it, the less I think a time-aware _initial_ split needs to consider `skip` or `order_by`, but I'm far from certain. I'd be happy to work on a PR to add a `shuffle` arg to `initial_split` if that seems right to you, or we can discuss scope for a new function with more features.

**Comment by ClaytonJY** [comment]
@topepo sure, are you open to a PR for that this week or should I wait to see what the other person is up to?

**Comment by topepo** [comment]
Go right ahead on `initial_time_split` if you want.
